### DISCUSSION TITLE
Does my site which is built on mkdocs support SEO automatically?

### DISCUSSION THREAD

**Discussion by mendax1234** [squidfunk/mkdocs-material#5932] [Q&A] [answered]
l want to know whether my site which is built on mkdocs material supports SEO automatically. Besides, how to improve my site's SEO, can l just use the general methods that can be got by google? 
Thanks!

**Comment by squidfunk** [comment, marked as answer]
Yes, Material for MkDocs goes to great lengths to provide a good basis for SEO. The site loads fast, works without JavaScript and tries to reduce content shift and other metrics measured by search engines like Google to rank the relevant and usability of a site. It mainly comes down to you to provide good content.

For instance, our own documentation ranks very good on Google, [often better than MkDocs itself](https://www.google.com/search?q=mkdocs+code+block).

**Reply by mendax1234** [reply]
l see. Thanks! But, are there any instructions on how to set the google SEO? Or l just need to write the content and don't care about anything because mkdocs material will do it for me automatically?

**Reply by squidfunk** [reply]
Write content and let Google do the rest. You could submit the `sitemap.xml` that MkDocs generates ([here's ours](https://squidfunk.github.io/mkdocs-material/sitemap.xml)) to Google webmaster tools to speed things up, but in general, you can just start writing.

**Reply by mendax1234** [reply]
Thanks a lot! How can l get the `sitemap.xml` generated by Mkdocs?

**Reply by squidfunk** [reply]
It's automatically generated.

**Reply by mendax1234** [reply]
Oh, l see it after using `mkdocs build`. Thanks a lot!

**Reply by raunaqss** [reply]
Hey, I have questions about this.

How can we add our ToC or Nav links to show up in Google's SERP snippet?

Thanks!!

**Reply by squidfunk** [reply]
I don't know, but if you find out, it would be great if you could share it here ☺️

**Reply by C3pa** [reply]
This page briefly overviews Google's SERP: https://www.highervisibility.com/seo/tools/serp-snippet-optimizer/ (I am in no way affiliated with that page). I guess browsing the Google documentation might provide a more in-depth overview of things that would need to be written in the HTML page to customize the page's SERP snippet.

Edit: I found them in Google docs: https://developers.google.com/search/docs/appearance/snippet

**Reply by squidfunk** [reply]
You can already set [the `description` of a page](https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-description) (as mentioned in the second link):

``` md
---
description: Nullam urna elit, malesuada eget finibus ut, ac tortor. 
---

# Page title
...
```

Other than that, please go into detail what Material for MkDocs should generate and what the upside is – how should snippets look? How would Google show them? I yet don't understand how this maps to documentation or static pages. The first link you posted is only a marketing site, so there's no information on it that is of use for us.

**Reply by raunaqss** [reply]
For some documentation pages where the DOM has semantic nav tags for the links in table of contents and an h3 tag for each nav link, they are visible on the google search results page itself like shown below:

<img width="618" alt="Screenshot 2024-07-31 at 12 35 13 PM" src="https://github.com/user-attachments/assets/8a5cfdcf-906b-45c7-82cc-ee829d315b16">

**Reply by alexvoss** [reply]
I am by no means an SEO expert but my impression is that Google decides in its own wisdom which pages get what. The only advice they seem to have is [the Sitelinks part of their documentation](https://developers.google.com/search/docs/appearance/sitelinks). If you look at the HTML produced by Material for MkDocs you should see that it already does a lot of relevant things right that fall into its scope. You as the site author can, of course, influence things by providing a sensible information architecture, writing in clear language, make labels relevant, etc.

**Reply by squidfunk** [reply]
@raunaqss you mean like this?

<img width="675" alt="Bildschirm­foto 2024-07-31 um 12 05 06" src="https://github.com/user-attachments/assets/91529c1c-b93a-4b60-8829-9e230509e843">
<img width="644" alt="Bildschirm­foto 2024-07-31 um 12 05 22" src="https://github.com/user-attachments/assets/bbd089e5-916e-4e08-adee-9d8cd69d9302">

**Reply by raunaqss** [reply]
Yes, my bad! I should've researched a bit more before asking here.

**Reply by squidfunk** [reply]
Well, those are live results from Google, so I guess there's nothing we need to add, right?

**Reply by squidfunk** [reply]
No worries, always good to double check from time to time, as Google changes their search results now and then ☺️
